pulp_rpm/app/modulemd.py

```diff
@@ -1,69 +1,80 @@
 import json
+import os
 import tempfile
 
-from django.db.utils import IntegrityError
 from pulpcore.app.models.content import Artifact
-from pulp_rpm.app.constants import PULP_MODULE_ATTR
+from pulp_rpm.app.constants import PULP_MODULE_ATTR, PULP_MODULEDEFAULTS_ATTR
 
 import gi
 gi.require_version('Modulemd', '2.0')
 from gi.repository import Modulemd as mmdlib  # noqa: E402
 
 
 def _create_snippet(snippet_string):
-    """Create snippet of modulemd[-defaults] as artifact."""
-    tmp_file = tempfile.NamedTemporaryFile()
-    with open(tmp_file.name, 'w') as f:
-        f.write(snippet_string)
-    artifact = Artifact.init_and_validate(tmp_file.name)
-    try:
-        artifact.save()
-    except IntegrityError:
-        artifact = Artifact.objects.get(sha256=artifact.sha256)
-    return artifact
+    """
+    Create snippet of modulemd[-defaults] as artifact.
 
+    Attributes:
+        snippet_string (string):
+            Snippet with modulemd[-defaults] yaml
 
-def get_modulemd_names(modulemd_yaml, module_index):
-    """Get modulemd names."""
-    ret, fails = module_index.update_from_string(modulemd_yaml, True)
-    if ret:
-        return module_index.get_module_names()
-    else:
-        return list()
+    Returns:
+        Snippet as unsaved Artifact object
+    """
+    tmp_file = tempfile.NamedTemporaryFile(dir=os.getcwd(), delete=False)
+    with open(tmp_file.name, 'w') as snippet:
+        snippet.write(snippet_string)
+    return Artifact.init_and_validate(tmp_file.name)
 
 
-def parse_modulemd(modules, module_index):
-    """Get modulemd NSVCA, artifacts, dependencies."""
+def parse_modulemd(module_names, module_index):
+    """
+    Get modulemd NSVCA, artifacts, dependencies.
+
+    Attributes:
+        module_names (list):
+            list of modulemd names
+        module_index (object):
```

Comment:
Can the type be `mmdlib.ModuleIndex`? Is it not a good idea for any reason?

Reply:
good idea, it will be more specific.